lua_examples/telnet/telnet.lua

```diff
@@ -143,11 +144,11 @@ return {
       wifi.setmode(wifi.STATION, false)
       wifi.sta.config { ssid = ssid, pwd  = pwd, save = false }
     end
-    tmr.alarm(0, 500, tmr.ALARM_AUTO, function()
+    local t = tmr.create()
+    t:alarm(500, tmr.ALARM_AUTO, function()
       if (wifi.sta.status() == wifi.STA_GOTIP) then
-        tmr.unregister(0)
-        print("Welcome to NodeMCU world", node.heap(), wifi.sta.getip())
-        net.createServer(net.TCP, 180):listen(port or 2323, telnet_listener)
+        t:unregister()
```

Comment:
It may be good practice, as @HHHartmann suggested for #2650, to add `t = nil` both here and in `ftpserver`.

Reply:
Added.
However I don't understand why is it not redundant. I thought that when the timer is unregistered, `t` will be gced at first opportunity.